### ISSUE TITLE
Log roll-by-time should have a limit of files option

### ISSUE THREAD

**Issue by maurokade** [kohsuke/winsw#182]
Hi guys, I think it would be helpful to limit the number of log files for roll-by-time too.

is it possible now?

**Comment by oleg-nenashev** [comment]
No, it's not possible now from what I see. It sounds reasonable to limit the depth of the history for this logging mode as well

**Comment by andrerfcsantos** [comment]
+1 to this. I think this is an essential feature to have, because with this you could effectively say "i only want to store logs for 5 days", but having each day in a separate file.

I may have to revert to `roll-by-size` just to limit the amount of logs generated. This is less than ideal because what i wanted was to limit them by number of days...

**Comment by keithneargarder** [comment]
New to this forum. This issue is a significant one for me also. Judging that it was reported 01/13/2017 and is still an issue 3+ years later I am guessing it is not going to get resolved. Very disappointing as it is a no-brainer to me - meaning an obviously useful feature not that it takes no intelligence to make the change. ;)

**Comment by NextTurn** [comment]
> Very disappointing as it is a no-brainer to me - meaning an obviously useful feature not that it takes no intelligence to make the change. ;)

The followings are steps to implement this no-brainer yourself that take no intelligence either:

- Fork this repository.
- Clone the forked repository.
`git clone https://github.com/keithneargarder/winsw`
- Create a branch.
`git checkout -b no-brainer`
- Make and commit changes.
`git commit`
- Push changes.
`git push origin no-brainer`
- Create a pull request.
https://github.com/keithneargarder/winsw/pull/new/no-brainer


Thanks advance for your contribution.

**Comment by keithneargarder** [comment]
> > Very disappointing as it is a no-brainer to me - meaning an obviously useful feature not that it takes no intelligence to make the change. ;)
> 
> The followings are steps to implement this no-brainer yourself that take no intelligence either:
> 
> * Fork this repository.
> * Clone the forked repository.
>   `git clone https://github.com/keithneargarder/winsw`
> * Create a branch.
>   `git checkout -b no-brainer`
> * Make and commit changes.
>   `git commit`
> * Push changes.
>   `git push origin no-brainer`
> * Create a pull request.
>   https://github.com/keithneargarder/winsw/pull/new/no-brainer
> 
> Thanks advance for your contribution.

Gee, thanks for such a warm welcome to the forum. I imagine I will be quite active on this forum with such inviting professionals such as yourself on this forum.

Thank you for your contribution to this issue as well. I will be sure to try your suggested fix.

BTW - since you obviously have a reading comprehension problem let me clarify. "No-brainer" like I said was in reference to the obvious usefulness and necessity of the "keepfiles" feature working as advertised. I said I recognize it DOES take intelligence to implement the fix.

Warmest of regards,
Keith

**Comment by A-Hassan-M** [comment]
any updates on this ?
